src/vs-picgo/index.ts

```diff
@@ -0,0 +1,192 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as os from 'os';
+import * as vscode from 'vscode';
+
+import PicGo from 'picgo/dist/core/PicGo';
+import { ImgInfo } from 'picgo/dist/utils/interfaces';
+
+import { promisify } from 'util';
+
+import { formatParam, formatString } from '../utils';
+
+const _ = require('lodash');
+const _db = require('lodash-id');
+_.mixin(_db);
+
+const writeFileP = promisify(fs.writeFile);
+const readFileP = promisify(fs.readFile);
+
+export interface INotice {
+  body: string;
+  text: string;
+  title: string;
+}
+
+export interface IUploadName {
+  date: string;
+  dateTime: string;
+  fileName: string;
+  extName: string;
+  mdFileName: string;
+  [key: string]: string;
+}
+
+export interface IOutputUrl {
+  uploadedName: string;
+  url: string;
+  [key: string]: string;
+}
+
+export default class VSPicgo {
+  private dataPath: string;
```

Comment:
We should not use `dataPath` as a field. If the user changed the data path settings, this field will not change and still be the old path when `vspicgo` was initialized. Always calling `getDataPath()` before logging to data is a good idea.

By the way, there is another bug needs to fix based on my test:
1. Upload a image using `Ctrl` + `Alt` + `U`
2. Update data path settings
3. Upload another image using `Ctrl` + `Alt` + `U`, and there will be neither success message showing up nor output inserted to the markdown file.

Reply:
> Code is quite clean, nice job.
> 
> Another 2 reminders:
> 
> 1. Change `picgo.picBed.qiniu.area` from enum to input and give a link to qiniu's official website.
> 2. We can add tests cases after this PR finished, cheers! (By the way, the `test` folder should be moved from under `src` to root for clarity.)
Test cases is a must. And the bug does show, i'll try to fix it.

Reply:
I accidentally discovered the cause of the problem:
<img src="https://picgo-1256492673.cos.ap-chengdu.myqcloud.com/img/20190426235030.png" alt="20190426235030.png">
We should get the editor again every time before we writing to the file.